public/js/test/mochitest/browser_dbg-navigation.js

```diff
@@ -0,0 +1,30 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+function countSources(dbg) {
+  const sources = dbg.selectors.getSources(dbg.getState());
+  return sources.size;
+}
+
+/**
+ * Test navigating
+ * navigating while paused will reset the pause state and sources
+ */
+add_task(function* () {
+  const dbg = yield initDebugger(
+    "doc-script-switching.html",
+    "script-switching-01.js"
+  );
+
+  invokeInTab("firstCall");
+  yield waitForPaused(dbg);
+
+  yield navigate(dbg, "doc-iframes.html");
+  yield waitForPaused(dbg);
+  assertPausedLocation(dbg, "iframes.html", 8);
+  ok(countSources(dbg) == 1);
```

Comment:
Might be nice to use `doc-scripts.html` which has a few more scripts. Testing with multiple scripts makes sure this isn't a fluke.